### ISSUE TITLE
Finding Parent for Children Return Immediately without Processing All Fields

### ISSUE THREAD

**Issue by sk0928** [FasterXML/jackson-databind#4252]
When finding parents of a property from a json, if a children has a property, it returns immediately, not parsing the whole json before recursion starts.
e.g. below structure return only one node in latest version however prior to 2.16, it return all 4.
```
{
    "type": "doc",
    "content": [
        {
            "type": "paragraph",
            "attrs": {
                "align": "left"
            },
            "content": [
                {
                    "type": "image",
                    "attrs": {
                        "id": "5b747f34-dea8-46d4-84b5-00aaa3a27cf7"
                    }
                },
                {
                    "type": "image",
                    "attrs": {
                        "id": "d372d49b-9996-4eb5-94ba-12371ce32df3"
                    }
                }
            ]
        }
    ]
}
```

https://github.com/FasterXML/jackson-databind/blob/36ecf8f38dbe3250a4a4f93a70340405c89cf838/src/main/java/com/fasterxml/jackson/databind/node/ObjectNode.java#L445

**Comment by pjfanning** [comment]
This is #4229

**Comment by cowtowncoder** [comment]
Yes. So will be fixed in 2.16.1.
